### ISSUE TITLE
Unregister default file: shortstop-handler

### ISSUE THREAD

**Issue by jknight12882** [krakenjs/kraken-js#470]
This handler wreaks havoc when you have a local_module installed in `package.json`. It tries to execute a read in `dependencies` and crashes out with

```
[ 'error',
  'EISDIR: illegal operation on a directory, read',
  { stack: 'Error: EISDIR: illegal operation on a directory, read\n    at Error (native)' } ]
error: EISDIR: illegal operation on a directory, read stack=Error: EISDIR: illegal operation on a directory, read
    at Error (native), @timestamp=2016-08-10T02:35:48.205Z, logname=application, type=-application-, severity=ERROR, sequencenumber=1, logmessage=EISDIR: illegal operation on a directory, read
Wed, 10 Aug 2016 02:35:48 GMT uncaughtException EISDIR: illegal operation on a directory, read
Error: EISDIR: illegal operation on a directory, read
    at Error (native)
```

**Comment by tlivings** [comment]
Hi there.

Could you provide some additional details about your configuration?

**Comment by jknight12882** [comment]
Here is the relevant portion of our `package.json`

```
"dependencies": {
    ...
    "react": "file:./local_modules/react",
    "react-dom": "file:./local_modules/react-dom",
   ...
  }
```

**Comment by grawk** [comment]
the `file` handler uses `fs.readFile`. I'd expect it wouldn't work for what you're attempting since I assume "react" and "react-dom" are actually directories.

**Comment by jknight12882** [comment]
that's exactly the point of this issue - this behavior is crashing my app. Since `kraken` by default registers the `file:` shortstop handler, and `npm` by default adds all the `package.json` config variables to the env, `krakenjs` is erroneously trying to parse the locale modules part of the `package.json` and throws an error

**Comment by jknight12882** [comment]
In other words, anybody using an npm supported `file:` dependency in their `package.json` is going to have a bad time

**Comment by grawk** [comment]
Oh, sorry. I didn't register that the file: directive was in your package.json as opposed to a config file...

**Comment by grawk** [comment]
There is currently no configurability around the set of shortstop handlers registered. That would seem to be the only resolution to this issue.

**Comment by tlivings** [comment]
Thanks for bringing this to our attention. Another workaround may be to ignore values pulled in from sources other than configuration during resolution.

**Comment by jknight12882** [comment]
Yeah, I was thinking the same thing - only parse things coming from the `config.json` for `shortstop-handlers`

**Comment by grawk** [comment]
Sounds like this issue should be filed against [confit](https://github.com/krakenjs/confit) instead?

**Comment by tlivings** [comment]
Package.json gets explicitly added here, not a confit concern.

**Comment by grawk** [comment]
I guess I need to reproduce this locally to fully understand what's happening... I'll give that a try tomorrow so I'm not talking out of my whatever.

**Comment by jknight12882** [comment]
@tlivings Where is `package.json` being explicitly added? The issue I was encountering was because `npm` adds `npm_config_*` environment variables for config values in `package.json`.

**Comment by grawk** [comment]
OK. I have it reproduced locally. You just have to add a local module into the package.json... which should be clear enough from @jknight12882 's comments already. But for reading comprehension on my part.

**Comment by grawk** [comment]
Submitted a PR to confit: https://github.com/krakenjs/confit/pull/65

This would give ability to configure a set of env properties for confit to ignore. Which in turn would prevent a shortstop-handler from attempting to process a string it shouldn't process.

The ignore property could then be set in kraken for the npm local_modules environment properties.

**Comment by grawk** [comment]
Made some changes to my fork of kraken-js to use the changes from the confit PR above:
https://github.com/grawk/kraken-js/commit/f3a946f73cb67d4c1142b5e70bea83ddfcb77ab7

This would merge the options passed into the kraken constructor under `options.confit` into the confit configurations used in kraken-js. So, if you used this:
```
var options = {
  confit: {
    envignore: ['npm_package_dependencies_react-dom', 'npm_package_dependencies_react']
  }
}
app.use(kraken(options));
```

Confit would simply ignore these environment variables and not attempt to match them to a shortstop handler.

**Comment by gshively11** [comment]
Hello @grawk! I find myself running into this issue nearly 7 years later. It seems like a fix was found, but didn't make it all the way into the main kraken-js project.  Will you accept a PR that implements the fix from your fork?

**Comment by grawk** [comment]
Hey @gshively11 .. happy to take a look and evaluate it. if it's an acceptable risk then we can move forward
